pkg/event/event.go

```diff
@@ -15,29 +15,24 @@ import (
 // GetAbnormalEventsForResource gets unique events of non-normal type for
 // a namespaced resource. Useful for logging the most relevant events
 // related to a resource we're waiting on.
+// Make sure to pass in the kind exactly as it is capitalized on the event, e.g. "Pod"
 func GetAbnormalEventsForResource(client client.Client,
 	nsName types.NamespacedName, resourceKind string) ([]corev1.Event, error) {
 	uniqueEventMap := make(map[string]corev1.Event)
 
 	eList := corev1.EventList{}
 	options := k8sclient.InNamespace(nsName.Namespace)
-	err := client.List(context.TODO(), options, &eList)
+	fieldSelector := fmt.Sprintf("involvedObject.name=%s,involvedObject.kind=%s,type!=Normal",
```

Comment:
Using a field selector here so that we don't have to pull back the full event list. Luckily, events are one of the fields it's possible to use field selectors on. See [this list](https://github.com/flant/shell-operator/blob/8fa3c3b8cfeb1ddb37b070b7a871561fdffe788b/HOOKS.md#fieldselector)

Reply:
NIT: Wonder if you can release the lines 24-30 with this:
```
		&k8sclient.ListOptions{
			Namespace:     nsName.Namespace,
			FieldSelector: fields.SelectorFromSet(fields.Set{"involvedObject.name": nsName.Name, "involvedObject.namespace": nsName.Namespace},
		}
```

Reply:
I don't think the field selector from set supports the `!=` operator, otherwise I would have used it.

Reply:
oh I totally missed the `!` part, +1